src/components/hosts/AdditionalNTPSourcesDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import {
+  Alert,
+  Form,
+  Modal,
+  ModalVariant,
+  ModalBoxBody,
+  ModalBoxFooter,
+  Button,
+  AlertActionCloseButton,
+  AlertVariant,
+  AlertActionLink,
+} from '@patternfly/react-core';
+import { Cluster, ClusterUpdateParams } from '../../api/types';
+import { ntpSourceValidationSchema } from '../ui/formik/validationSchemas';
+import TextAreaField from '../ui/formik/TextAreaField';
+import { patchCluster } from '../../api/clusters';
+import { updateCluster } from '../../features/clusters/currentClusterSlice';
+import { getErrorMessage, handleApiError } from '../../api/utils';
+import { trimCommaSeparatedList } from '../ui/formik/utils';
+
+type AdditionalNTPSourcesDialogToggleProps = {
+  cluster: Cluster;
+  onToggle?: (isOpen: boolean) => void;
+};
+
+export const AdditionalNTPSourcesDialogToggle: React.FC<AdditionalNTPSourcesDialogToggleProps> = ({
+  cluster,
+  onToggle,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const setOpen = (isOpen: boolean) => {
+    onToggle && onToggle(isOpen);
```

Comment:
Then this would be simplified

Reply:
I know that the field is currently used but there is a possibility that in the future the dialog is going to be open from somewhere else than popover and onToggle won't be necessary. I'd like to keep it as optional to avoid ( my :sweat_smile:  ) confusion later.